sim/core/attack.go

```diff
@@ -260,12 +263,76 @@ func (wa *WeaponAttack) setWeapon(weapon Weapon) {
 // inlineable stub for swing
 func (wa *WeaponAttack) trySwing(sim *Simulation) time.Duration {
 	if sim.CurrentTime < wa.swingAt {
+
+		if sim.Log != nil {
+			wa.unit.Log(sim, "trySwing failed - swing at %f and lastswing at %f", wa.swingAt, wa.lastSwingAt)
+		}
 		return wa.swingAt
 	}
 	return wa.swing(sim)
 }
 
+func (wa *WeaponAttack) castExtraAttacksStored(sim *Simulation) {
+	wa.castExtraAttacks(sim, &wa.extraAttacksStored, 0)
+	wa.extraAttacksStored = 0;
+
+	if wa.extraAttacksAura != nil {
+		wa.extraAttacksAura.SetStacks(sim, 0)
+	}
+}
+
+func (wa *WeaponAttack) castExtraAttacksTriggered(sim *Simulation, moreAttacks bool) {
+	if moreAttacks {
+		wa.castExtraAttacks(sim, &wa.extraAttacks, 0)
+	} else {
+		wa.castExtraAttacks(sim, &wa.extraAttacks, 1)
+	}
+}
+
+func (wa *WeaponAttack) castExtraAttacks(sim *Simulation, numExtraAttacks *int32, startIndex int32) bool {
+	if *numExtraAttacks > 0 {
+		// Ignore the first extra attack, that was used to speed up next attack
+		/*if sim.Log != nil {
+			wa.unit.Log(sim, "Setting MH spell metric to 1")
+		}*/
```

Comment:
Could you clear up some of these extra commented out logs that are leftover here and in some of the other files?

Reply:
cleaned up